internal/testutil/testutil.go

```diff
@@ -0,0 +1,33 @@
+package testutil
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"testing"
+)
+
+// TestSkipUnlessCI skips the current test with a provided message, except when running
+// in CI environment, in which case it causes hard failure.
+func TestSkipUnlessCI(t *testing.T, msg string, args ...interface{}) {
+	t.Helper()
+
+	if len(args) > 0 {
+		msg = fmt.Sprintf(msg, args...)
+	}
+
+	if os.Getenv("CI") != "" {
+		t.Fatal(msg)
+	} else {
+		t.Skip()
+	}
```

Comment:
nit: Is the intent to pass `msg` to `t.Skip()` as well? If not, why not skip and bail out early when appropriate?


```suggestion
	if os.Getenv("CI") == "" {
		t.Skip()
	} 

	if len(args) > 0 {
		msg = fmt.Sprintf(msg, args...)
	}

	t.Fatal(msg)
```

Reply:
fixed